File: 3052.c
#include <stdio.h>

int main(void) {

	int num_list[42] = { 0 };
	int nums[10];
	int count = 0;


	for (int i = 0; i < 10; i++) {

		int plate;

		scanf("%d", &plate);
		nums[i] = plate;

	}

	for (int i = 0; i < 10; i++) {

		int plate = nums[i] % 42;
		num_list[plate]++;

	}

	for (int i = 0; i < 42; i++) {

		if (num_list[i] != 0) {

			count++;

		}

	}

	printf("%d", count);

}

File: Python/3052.py
num_list = []

for i in range(0,42):
    num_list[i] = 0
    
for i in range(0,10):
    index = int(input())
    index = index % 42
    num_list[index] += 1

"""count = 0;

for i in num_list:

    if i != 0:
        count += 1
"""
print(len(num_list))
